streams/src/main/java/org/apache/kafka/streams/state/internals/ChangeLoggingWindowBytesStore.java

```diff
@@ -76,6 +76,10 @@
 
     @Override
     public void put(final Bytes key, final byte[] value) {
+        // Note: It's incorrect to bypass the wrapped store here by delegating to another method,
+        // but we have no alternative. We must send a timestamped key to the changelog, which means
+        // we need to know what timestamp gets used for the record. Hopefully, we can deprecate this
+        // method in the future to resolve the situation.
         put(key, value, context.timestamp());
```

Comment:
In this case, as the comment explains, calling across to the other method is the best option for us. Really, it just demonstrates that this method should be deprecated (https://issues.apache.org/jira/browse/KAFKA-7928), at which point the problem goes away.